src/attributes.rs

```diff
@@ -316,6 +390,9 @@ pub fn parse_segment_attributes(attrs: &[Attribute]) -> SegmentAttributes {
             ParsedAttribute::ConstantAccess(_) => {
                 panic!("Const attribute cannot be specified on a `to <expr>` segment.");
             }
+            ParsedAttribute::Expr(_) => {
```

Comment:
See my other comment about what should `$expr` expand to. In that context, I think that we should allow this to be a segment attribute, as that could be quite useful.